### PR TITLE
Refactor some notifications to use consolidated dispatcher in Admin and Dashboards

### PR DESCRIPTION
**Pull request by jaredscheib**
- [x] Rebased/mergable
  - [x] Tests pass

Connect #1114 

### The problem
- Inconsistent use of legacy `addFlashMessage` vs `publishNotifications`.
- Actions were performing multiple dispatches to create auto-dismissing notification effect.
- Some AJAX requests were performing notification dispatch.
- `deleteUserAsync` and `deleteRoleAsync` were written out of async/await/try/catch convention.
- Constants were being used out of scope.

### The Solution
- Create a shared `publishAutoDismissingNotification` dispatcher.
- Remove use of `addFlashMessage` in throughout Admin and Dashboards.
- Refactor `deleteUserAsync` and `deleteRoleAsync` to be consistent with latest convention.
- Consolidate and share constants.

### CODE REVIEW

File: ui/src/admin/actions/index.js [hunk 1]
```diff
@@ -20,8 +20,10 @@ import {
   killQuery as killQueryProxy,
 } from 'shared/apis/metaQuery'
 
-import {publishNotification} from 'src/shared/actions/notifications';
-import {ADMIN_NOTIFICATION_DELAY} from 'src/admin/constants'
+import {publishNotification} from 'shared/actions/notifications';
```
Comment (cryptoquick) [inline ui/src/admin/actions/index.js]: 23:65 error  Extra semicolon  semi

File: ui/src/admin/actions/index.js [hunk 2]
```diff
@@ -306,25 +308,29 @@ export const killQueryAsync = (source, queryID) => (dispatch) => {
...
-export const deleteRoleAsync = (role, addFlashMessage) => (dispatch) => {
-  // optimistic update
+export const deleteRoleAsync = (role) => async (dispatch) => {
   dispatch(deleteRole(role))
-
-  // delete role on server
-  deleteRoleAJAX(role.links.self, addFlashMessage, role.name)
+  dispatch(publishAutoDismissingNotification('success', 'Role deleted'))
+  try {
+    await deleteRoleAJAX(role.links.self)
```
Comment (lukevmorris) [inline ui/src/admin/actions/index.js]: I wonder if the notification should come after this call -- if we report it as deleted, but the AJAX request fails, then we'll publish another notification that we failed to delete it

File: ui/src/admin/actions/index.js [hunk 3]
```diff
@@ -306,25 +308,29 @@ export const killQueryAsync = (source, queryID) => (dispatch) => {
...
 }
 
-export const deleteUserAsync = (user, addFlashMessage) => (dispatch) => {
-  // optimistic update
+export const deleteUserAsync = (user) => async (dispatch) => {
   dispatch(deleteUser(user))
-
-  // delete user on server
-  deleteUserAJAX(user.links.self, addFlashMessage, user.name)
+  dispatch(publishAutoDismissingNotification('success', 'User deleted'))
```
Comment (lukevmorris) [inline ui/src/admin/actions/index.js]: Does this come after the AJAX call?

**Comment by jaredscheib** [comment]
@cryptoquick killed the semis and merged in updated master

File: ui/src/admin/actions/index.js (hunk 2 above)
Comment (jaredscheib) [inline ui/src/admin/actions/index.js]: yeah, good call. submitted the changes. the thought was that there's a trade-off. currently we're performing optimistic updates (aka updating state as though AJAX will be successful), which makes when to display the notification trickier: either 1) it notifies success at the same time as deletion of the item in the UI, but then an error message could come up shortly thereafter, which is confusing; or, 2) it removes the item from the UI, and notifies success upon AJAX success, which prevents showing both success and failure notifications, but also introduces a delay between UI state change and when the notification will present. thoughts?

**Comment by lukevmorris** [comment]
@jaredscheib answered a couple of my questions:
- How come we don't submit the AJAX request first and proceed to changing Redux if successful?
Because our convention is to update optimistically, and communicate a failure down the road.
- How come we don't handle AJAX failure in a couple of spots in this PR? This is a known issue (#997), which is out of scope for this PR. We will address soon.

With the above info, I think this PR is good to go.

**Pull request merged**